### PR TITLE
More log options for k8s 1.6

### PR DESCRIPTION
**Pull request by justinsb**
Follow on to #2095

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2202)

### CODE REVIEW

File: nodeup/pkg/model/logrotate.go [hunk 1]
```diff
@@ -43,5 +46,70 @@ func (b *LogrotateBuilder) Build(c *fi.ModelBuilderContext) error {
...
+	if err != nil || k8sVersion == nil {
+		return fmt.Errorf("unable to parse KubernetesVersion %q", b.Cluster.Spec.KubernetesVersion)
+	}
+
+	if k8sVersion.Major == 1 && k8sVersion.Minor < 6 {
+		// In version 1.6, we move log rotation to docker, but prior to that we need a logrotate rule
+		b.addLogRotate(c, "docker-containers", "/var/lib/docker/containers/*/*-json.log", logRotateOptions{MaxSize: "10M"})
+	}
+
+	b.addLogRotate(c, "docker", "/var/log/docker.log", logRotateOptions{})
```
Comment (kris-nova) [inline nodeup/pkg/model/logrotate.go]: Do we want logrotate on `/var/log/messages` at all? Just curious (I think EBS can spam this hard)

File: upup/pkg/fi/resources.go [hunk 2]
```diff
@@ -119,6 +119,10 @@ type StringResource struct {
 	s string
 }
 
+func (r *StringResource) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&r.s)
```
Comment (kris-nova) [inline upup/pkg/fi/resources.go]: Any interest in `fmt.Errorf()`ing this json error?

File: nodeup/pkg/model/logrotate.go (hunk 1 above)
Comment (justinsb) [inline nodeup/pkg/model/logrotate.go]: Possibly - and we might already be doing that by default as it is a "standard log file".  If you open an issue we can take a look!

File: upup/pkg/fi/resources.go (hunk 2 above)
Comment (justinsb) [inline upup/pkg/fi/resources.go]: I think this one is deep in the json marshalling machinery and I don't know what magic errors is uses....

I actually added this one just so the output would appear in the integration test output ... we're not actually using it per-se.

**Pull request merged**